core/compiler.cpp

```diff
@@ -180,8 +184,33 @@ partitioning::GraphAndMapping BuildHybridGraph(
             true);
 
         seg_block.update_graph(temp_g);
+      } else {
+        num_torch_segments++;
+
+        // If full compilation is expected, ensure that all operators in Torch blocks are
+        // for collections processing
+        if (expect_full_compilation) {
+          for (auto torch_node : seg_block.block()->nodes()) {
+            if (partitioning::CollectionNodeKinds.find(torch_node->kind()) == partitioning::CollectionNodeKinds.end()) {
+              TORCHTRT_THROW_ERROR(
+                  "Full compilation specified but node " << torch_node->kind().toQualString()
+                                                         << " was executed in Torch."
+                                                         << " Try recompiling with require_full_compilation=False.");
+            }
+          }
+        }
       }
     }
+
+    // If full compilation is expected, cannot have more than 2 Torch segments
+    // (one for preprocessing inputs, one for post-processing outputs) and 1 TRT segment
+    if (expect_full_compilation && !(num_torch_segments <= 2 && num_trt_segments == 1)) {
+      TORCHTRT_THROW_ERROR(
+          "Full compilation specified but number of torch segments was "
+          << num_torch_segments << " and number of trt segments was " << num_trt_segments
+          << ". Was expecting at most 2 Torch segments and 1 TRT segment."
+          << " Try recompiling with require_full_compilation=False.");
+    }
```

Comment:
This error should be something like: `Full compilation was requested but unable to convert all operations to TensorRT. Try recompile with require_full_compilation=False`